compiler/fir/checkers/src/org/jetbrains/kotlin/fir/analysis/checkers/syntax/FirAnonymousFunctionSyntaxChecker.kt

```diff
@@ -46,4 +48,7 @@ object FirAnonymousFunctionSyntaxChecker : FirDeclarationSyntaxChecker<FirAnonym
             )
         }
     }
+
+    override fun isApplicable(element: FirAnonymousFunction, source: KtSourceElement): Boolean =
```

Comment:
Why is it necessary?

Reply:
This change was required after all nodes that are created in `toBuildLiteralCall` had the appropriate fake source element (see [31ec0cb](https://github.com/JetBrains/kotlin/pull/4803/commits/31ec0cb04bde769ec4d5e6c9699a3359582cd2b9))

It led to the fact that the tests began to fail with the `ClassCastException` which is throwing [from there](https://github.com/DMozhevitin/kotlin/blob/master/compiler/fir/checkers/src/org/jetbrains/kotlin/fir/analysis/checkers/syntax/FirSyntaxChecker.kt#L33):

```kotlin
org.jetbrains.kotlin.psi.KtStringTemplateExpression cannot be cast to org.jetbrains.kotlin.psi.KtFunction
java.lang.ClassCastException: org.jetbrains.kotlin.psi.KtStringTemplateExpression cannot be cast to org.jetbrains.kotlin.psi.KtFunction
```

As a reference, I looked how this problem is solved in case of `for` loop: it's done [like this](https://github.com/DMozhevitin/kotlin/blob/master/compiler/fir/checkers/src/org/jetbrains/kotlin/fir/analysis/checkers/syntax/FirLocalVariableTypeParametersSyntaxChecker.kt#L25), so I used the same approach, but I'm still not completely sure that it's a good solution.